From ff4d0f87da9b7961e8e69e0f77ed25b8b7569755 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 22 Apr 2013 18:39:23 +0300
Subject: [PORT FROM MAIN] Telephony: remove setCurrentPreferredNetworkType done on SIM ready

BZ: 45846

Upon SIM ready, RIL_REQUEST_SET_PREFERRED_NETWORK_TYPE
is sent to make sure the network type is set for modem
which does not restore the type. setPreferredNetworkType
is already issued upon RIL_UNSOL_RIL_CONNECTED message.
Doing it again in SIM ready is redundant.

Also setting preferred network type on SIM ready,
will result in emergency call being dropped if the
user unlocks the phone when the emergency call is connected.

This patch removes the setCurrentPreferredNetworkType done
on SIM ready.

Change-Id: I6496ce1f89ce351a5c2ea7105868fb8a8c2c8b48
Orig-Change-Id: Id0c4562218f1c12b23137f2909f77adfc443f34b
Category: device enablement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/gsm/GsmServiceStateTracker.java      |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index ef42b24..3d257a7 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -288,8 +288,6 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
                 break;
 
             case EVENT_SIM_READY:
-                // Set the network type, in case the radio does not restore it.
-                mCi.setCurrentPreferredNetworkType();
 
                 boolean skipRestoringSelection = mPhone.getContext().getResources().getBoolean(
                         com.android.internal.R.bool.skip_restoring_network_selection);
-- 
1.7.4.1

